app/graph/tracers/stdout_tracer.rb

```diff
@@ -0,0 +1,10 @@
+class StdoutTracer
+  def self.trace(key, data)
```

Comment:
I've added this so we can use it for some debugging, it has been quite useful to me but isn't part of this PR.